stratum/hal/lib/barefoot/bfrt_p4runtime_translator.cc

```diff
@@ -683,6 +685,9 @@ ::util::StatusOr<std::string> BfrtP4RuntimeTranslator::TranslateTnaPortId(
     const uint32 sdk_port_id = singleton_port_to_sdk_port_[port_id];
     std::string sdk_port_id_bytes = P4RuntimeByteStringToPaddedByteString(
         Uint32ToByteStream(sdk_port_id), NumBitsToNumBytes(bit_width));
+    if (!FLAGS_incompatible_enable_bfrt_legacy_bytestring_responses) {
+      sdk_port_id_bytes = ByteStringToP4RuntimeByteString(sdk_port_id_bytes);
```

Comment:
just want to confirm, we also want to make the byte string canonical when writing to the SDK, right?

Reply:
Probably not. The Wrapper will just pad it again. Good catch.